### PR TITLE
New DiscoverDcType prop

### PR DESCRIPTION
**Pull request by DavidePrincipi**
It controls the way LDAP clients, Samba and SSSD find an Active Directory
domain controller. 

NethServer/dev#5534

### CODE REVIEW

**Comment by nrauso** [comment]
Whoa! Queries are now so much faster!

File: root/etc/e-smith/templates/etc/samba/smb.conf/10base [hunk 1]
```diff
@@ -7,6 +7,11 @@ server string = NethServer {$sysconfig{'Version'} . ' ' . $sysconfig{'Release'}}
 security = { $sssd{'Provider'} eq 'ad' ? 'ADS' : 'user' }
 { $sssd{'Provider'} eq 'ad' ? '' : '# ' }realm = { $sssd{'Realm'} }
 { $sssd{'Provider'} eq 'ad' ? '' : '# ' }kerberos method = secrets and keytab
+{ $sssd{'Provider'} eq 'ad' ? '' : '# ' }password server = {
```
Comment (gsanchietti) [inline root/etc/e-smith/templates/etc/samba/smb.conf/10base]: The default value for `password server` should be `*`.

From `man smb.conf`:

> Ideally, this option should not be used, as the default '*' indicates to Samba to determine the best DC to contact dynamically, just as all other hosts in an AD domain do. This allows the domain
           to be maintained (addition and removal of domain controllers) without modification to the smb.conf file.

Comment (DavidePrincipi) [reply root/etc/e-smith/templates/etc/samba/smb.conf/10base]: Also, the allowed prop values should be reduced to two:
- use DNS to find the DC address
- use a fixed DC address

**Comment by DavidePrincipi** [comment]
Already tested ldapuri in a multi-dc environment

**Pull request merged**